scanner.go

```diff
@@ -241,13 +213,12 @@ func (s *scanner) request() (*pb.ScanResponse, hrpc.RegionInfo, error) {
 
 // update updates the scanner for the next scan request
 func (s *scanner) update(resp *pb.ScanResponse, region hrpc.RegionInfo) {
-	if resp.GetMoreResultsInRegion() {
-		if resp.ScannerId != nil {
-			s.scannerID = resp.GetScannerId()
-		}
-	} else {
+	if s.isRegionScannerClosed() && resp.ScannerId != nil {
```

Comment:
so we don't want to update scanner ID every time? I guess we don't want to rely on HBASE-18000 as it's not desired way?

Reply:
This is to make sure we have closed the scanner on last region and this is for a new region. As client, we cannot update scanner id if not returned from server.

Reply:
I also removed the check for resp.GetMoreResultsInRegion() because we need the scanner id for the call to close region scanner.